daemon/src/socket.rs

```diff
@@ -9,8 +11,9 @@ use tokio_util::codec::{FramedWrite, LengthDelimitedCodec};
 #[derive(Debug, Serialize, Deserialize)]
 #[serde(tag = "type", content = "payload")]
 pub enum Message {
-    Ping,
-    Pong,
+    SendCurrentOffer(CfdOffer),
+    TakeOffer(CfdTakeRequest),
+    ConfirmOffer(Cfd),
 }
```

Comment:
Perhaps we can split this enum into two:

```rust
enum MakerToTaker {

}

enum TakerToMaker {

}
```

If you make `spawn_sender` generic over the message, this should work.